### ISSUE TITLE
Post upgrade 3.1.31 to 4.1.3: search index can't be recreated.

### ISSUE THREAD

**Issue by yacdaniel** [craftercms/craftercms#6610]
### Duplicates

- [X] I have searched the existing issues

### Latest version

- [x] The issue is in the develop branch
- [X] The issue is in the latest released 4.1.x
- [ ] The issue is in the latest released 4.0.x
- [ ] The issue is in the latest released 3.1.x

### Describe the issue

There is an error during the re create of index in the `post-upgrade.sh` step. It only affects Authoring, Delivery is upgrading fine. 

### Steps to reproduce

Steps:

1. Start up Authoring 3.1.31 and crearte a couple of sitres. 
2. Run the upgrade scripts `upgrade-search.sh`, `upgrade-target.sh`  and `post-upgrade.sh`
3. See the issue

### Relevant log output

```shell
Deployer log
2024-03-08 14:05:36.514  INFO 30207 --- [http-nio-9191-exec-6] ensearch.impl.OpenSearchAdminServiceImpl : Recreating index for alias wordify123-authoring
2024-03-08 14:05:36.517  INFO 30207 --- [http-nio-9191-exec-6] ensearch.impl.OpenSearchAdminServiceImpl : Found index wordify123-authoring_reindexed_710_v2 for alias wordify123-authoring
2024-03-08 14:05:36.518  INFO 30207 --- [http-nio-9191-exec-6] ensearch.impl.OpenSearchAdminServiceImpl : Found locale authoring_REINDEXED_710 for index wordify123-authoring_reindexed_710_v2
2024-03-08 14:05:36.521 ERROR 30207 --- [http-nio-9191-exec-6] cms.deployer.impl.rest.ExceptionHandlers : Request /api/1/target/recreate/authoring/wordify123 failed with status 500 INTERNAL_SERVER_ERROR
org.craftercms.search.opensearch.exception.OpenSearchException: [wordify123-authoring] Error upgrading index wordify123-authoring
	at org.craftercms.search.opensearch.impl.OpenSearchAdminServiceImpl.doRecreateIndex(OpenSearchAdminServiceImpl.java:293)
	at org.craftercms.search.opensearch.impl.OpenSearchAdminServiceImpl.recreateIndex(OpenSearchAdminServiceImpl.java:249)
	at org.craftercms.deployer.impl.TargetServiceImpl.recreateIndex(TargetServiceImpl.java:289)
	at org.craftercms.deployer.impl.rest.TargetController.recreateIndex(TargetController.java:336)
	at org.craftercms.deployer.impl.rest.TargetController$$FastClassBySpringCGLIB$$b2995acb.invoke(<generated>)
	at org.springframework.cglib.proxy.MethodProxy.invoke(MethodProxy.java:218)
	at org.springframework.aop.framework.CglibAopProxy$CglibMethodInvocation.invokeJoinpoint(CglibAopProxy.java:792)
	at org.springframework.aop.framework.ReflectiveMethodInvocation.proceed(ReflectiveMethodInvocation.java:163)
	at org.springframework.aop.framework.CglibAopProxy$CglibMethodInvocation.proceed(CglibAopProxy.java:762)
	at org.springframework.validation.beanvalidation.MethodValidationInterceptor.invoke(MethodValidationInterceptor.java:123)
	at org.springframework.aop.framework.ReflectiveMethodInvocation.proceed(ReflectiveMethodInvocation.java:186)
	at org.springframework.aop.framework.CglibAopProxy$CglibMethodInvocation.proceed(CglibAopProxy.java:762)
	at org.springframework.aop.framework.CglibAopProxy$DynamicAdvisedInterceptor.intercept(CglibAopProxy.java:707)
	at org.craftercms.deployer.impl.rest.TargetController$$EnhancerBySpringCGLIB$$42fc37e6.recreateIndex(<generated>)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:77)
	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.base/java.lang.reflect.Method.invoke(Method.java:568)
	at org.springframework.web.method.support.InvocableHandlerMethod.doInvoke(InvocableHandlerMethod.java:205)
	at org.springframework.web.method.support.InvocableHandlerMethod.invokeForRequest(InvocableHandlerMethod.java:150)
	at org.springframework.web.servlet.mvc.method.annotation.ServletInvocableHandlerMethod.invokeAndHandle(ServletInvocableHandlerMethod.java:117)
	at org.springframework.web.servlet.mvc.method.annotation.RequestMappingHandlerAdapter.invokeHandlerMethod(RequestMappingHandlerAdapter.java:898)
	at org.springframework.web.servlet.mvc.method.annotation.RequestMappingHandlerAdapter.handleInternal(RequestMappingHandlerAdapter.java:809)
	at org.springframework.web.servlet.mvc.method.AbstractHandlerMethodAdapter.handle(AbstractHandlerMethodAdapter.java:87)
	at org.springframework.web.servlet.DispatcherServlet.doDispatch(DispatcherServlet.java:1072)
	at org.springframework.web.servlet.DispatcherServlet.doService(DispatcherServlet.java:965)
	at org.springframework.web.servlet.FrameworkServlet.processRequest(FrameworkServlet.java:1006)
	at org.springframework.web.servlet.FrameworkServlet.doPost(FrameworkServlet.java:909)
	at javax.servlet.http.HttpServlet.service(HttpServlet.java:555)
	at org.springframework.web.servlet.FrameworkServlet.service(FrameworkServlet.java:883)
	at javax.servlet.http.HttpServlet.service(HttpServlet.java:623)
	at org.apache.catalina.core.ApplicationFilterChain.internalDoFilter(ApplicationFilterChain.java:209)
	at org.apache.catalina.core.ApplicationFilterChain.doFilter(ApplicationFilterChain.java:153)
	at org.apache.tomcat.websocket.server.WsFilter.doFilter(WsFilter.java:51)
	at org.apache.catalina.core.ApplicationFilterChain.internalDoFilter(ApplicationFilterChain.java:178)
	at org.apache.catalina.core.ApplicationFilterChain.doFilter(ApplicationFilterChain.java:153)
	at org.springframework.web.filter.RequestContextFilter.doFilterInternal(RequestContextFilter.java:100)
	at org.springframework.web.filter.OncePerRequestFilter.doFilter(OncePerRequestFilter.java:117)
	at org.apache.catalina.core.ApplicationFilterChain.internalDoFilter(ApplicationFilterChain.java:178)
	at org.apache.catalina.core.ApplicationFilterChain.doFilter(ApplicationFilterChain.java:153)
	at org.springframework.web.filter.FormContentFilter.doFilterInternal(FormContentFilter.java:93)
	at org.springframework.web.filter.OncePerRequestFilter.doFilter(OncePerRequestFilter.java:117)
	at org.apache.catalina.core.ApplicationFilterChain.internalDoFilter(ApplicationFilterChain.java:178)
	at org.apache.catalina.core.ApplicationFilterChain.doFilter(ApplicationFilterChain.java:153)
	at org.springframework.web.filter.CharacterEncodingFilter.doFilterInternal(CharacterEncodingFilter.java:201)
	at org.springframework.web.filter.OncePerRequestFilter.doFilter(OncePerRequestFilter.java:117)
	at org.apache.catalina.core.ApplicationFilterChain.internalDoFilter(ApplicationFilterChain.java:178)
	at org.apache.catalina.core.ApplicationFilterChain.doFilter(ApplicationFilterChain.java:153)
	at org.apache.catalina.core.StandardWrapperValve.invoke(StandardWrapperValve.java:168)
	at org.apache.catalina.core.StandardContextValve.invoke(StandardContextValve.java:90)
	at org.apache.catalina.authenticator.AuthenticatorBase.invoke(AuthenticatorBase.java:481)
	at org.apache.catalina.core.StandardHostValve.invoke(StandardHostValve.java:130)
	at org.apache.catalina.valves.ErrorReportValve.invoke(ErrorReportValve.java:93)
	at org.apache.catalina.core.StandardEngineValve.invoke(StandardEngineValve.java:74)
	at org.apache.catalina.connector.CoyoteAdapter.service(CoyoteAdapter.java:346)
	at org.apache.coyote.http11.Http11Processor.service(Http11Processor.java:390)
	at org.apache.coyote.AbstractProcessorLight.process(AbstractProcessorLight.java:63)
	at org.apache.coyote.AbstractProtocol$ConnectionHandler.process(AbstractProtocol.java:928)
	at org.apache.tomcat.util.net.NioEndpoint$SocketProcessor.doRun(NioEndpoint.java:1794)
	at org.apache.tomcat.util.net.SocketProcessorBase.run(SocketProcessorBase.java:52)
	at org.apache.tomcat.util.threads.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1191)
	at org.apache.tomcat.util.threads.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:659)
	at org.apache.tomcat.util.threads.TaskThread$WrappingRunnable.run(TaskThread.java:63)
	at java.base/java.lang.Thread.run(Thread.java:840)
Caused by: org.craftercms.search.commons.exception.IndexNotFoundException: Target 'wordify123-authoring_reindexed_710_v2' not found
	at org.craftercms.search.opensearch.impl.OpenSearchAdminServiceImpl.doGetIndexSettings(OpenSearchAdminServiceImpl.java:318)
	at org.craftercms.search.opensearch.impl.OpenSearchAdminServiceImpl.doRecreateIndex(OpenSearchAdminServiceImpl.java:277)
	... 63 more

> curl http://localhost:9201/_cat/indices
yellow open ed-authoring_reindexed_710_v2    12_WMvWPQfqgjDkc31DHvA 1 1 72 4 402.8kb 402.8kb
green  open .opensearch-observability        apojusZnTF6rMH68JX7SxA 1 0  0 0    208b    208b
yellow open ed-preview_reindexed_710_v2      B3-6mdNfSQOtyUXyJGR-Qw 1 1 28 0 264.6kb 264.6kb
yellow open .plugins-ml-config               dSsNi0seTsCjP_xlqcQv4g 1 1  1 0   3.8kb   3.8kb
yellow open store-authoring_reindexed_710_v2 b6eVVwvyTxeQX4cpXuYvVA 1 1 28 0  89.3kb  89.3kb
yellow open store-preview_reindexed_710_v2   84olyIcnSgKodo2vaqnHBw 1 1 13 0  70.8kb  70.8kb
green  open .tasks                           QMBJyu7TTPSlSYBb2A9zew 1 0  4 0  27.4kb  27.4kb
```

**Comment by jmendeza** [comment]
This bug has to do with the recreate index API
Please re-test related operations besides the upgrade: `api/1/target/recreate/:env/:site_name`  API and duplicate site

**Comment by yacdaniel** [comment]
Verified fixed in `4.1.3-SNAPSHOT`

**Comment by yacdaniel** [comment]
Verified in `4.2.0-SNAPSHOT`
